Modify how ROSA is detected as platform and move from cloud_type to platform

morenod opened the pull request:
ROSA was failing to be detected as platform, as it is not always the first tag on [utils/common.sh](https://github.com/cloud-bulldozer/e2e-benchmarking/blob/master/utils/common.sh#L128), for example:

```
        "platformStatus": {
            "aws": {
                "region": "us-west-2",
                "resourceTags": [
                    {
                        "key": "red-hat-managed",
                        "value": "true"
                    },
                    {
                        "key": "red-hat-clustertype",
                        "value": "rosa"
                    }
                ]
            },
            "type": "AWS"
        }
    }
}
```

Current return: true, instead of rosa


Also moving ROSA installation to PLATFORMs instead of CLUSTERTYPE to allow comparing common installations to managed services ones.



Example executing ingress-performance and uploaded to perf-results-elastic.apps.observability.perfscale.devcluster.openshift.com:80

Workload finished, results:
{
    "termination": "http",
    "test_type": "http",
    "uuid": "fe3194fd-7a41-412a-82cb-a74fc8b342f4",
    "cluster.id": "a8c99ffd-2b59-4680-8e67-433cceffa423",
    "cluster.name": "mrnd-timing-nw4w9",
    "cluster.ocp_version": "4.10.6",
    "cluster.kubernetes_version": "1.23",
    "cluster.sdn": "OVNKubernetes",
    "cluster.platform": "ROSA",
    "requests_per_second": 44207,
    "avg_latency": 2258,
    "latency_95pctl": 3726,
    "latency_99pctl": 7183,
    "host_network": "true",
    "sample": "1",
    "runtime": 60,
    "routes": 100,
    "conn_per_targetroute": 1,
    "keepalive": 0,
    "tls_reuse": true,
    "number_of_routers": "2",
    "200": 2652467
}

jaredoconnell commented:
My fix for the first problem listed was merged in 20 days ago. Was there something wrong with that fix?

morenod commented:
> My fix for the first problem listed was merged in 20 days ago. Was there something wrong with that fix?

On current code, we are selecting the first tag:

.status.platformStatus.aws.resourceTags[0].value

But it is not always fixed to be the first tag, sometimes it is the second one, so executing current code on one of my clusters, the result of that jq is "true" instead of "rosa", because it is accessing "red-hat-managed" tag instead of red-hat-clustertype

mkarg75 commented:
Back at the initial version, I asked David if the detection works the way he wanted it, so we went with it. Turns out that the test scenario I used made us miss a variation, so be it. 
LGTM otherwise.

morenod commented:
> Back at the initial version, I asked David if the detection works the way he wanted it, so we went with it. Turns out that the test scenario I used made us miss a variation, so be it. LGTM otherwise.

Yep, it was working, but when creating the dashboards, I realised that having a cluster type specific for rosa dont let us to compare with other platforms, so I have removed all references to cluster type and moved rosa to a platform.

Code now is also prepared to be easy to integrate new managed services platforms that will come shortly

jaredoconnell commented:
I see. So there was some duplicate code before, with some of it still having the bug.

This new code works the same way my old code worked, so looks good to me.

dry923 commented:
@morenod I'm fine with this for rosa however this same process wont work for osd as it does not apply a flag that I can find (at least not on GCP)

dry923 commented:
capturing my comment from slack:
"
yeah honestly not loving the web in those files. not your fault just an observation. that said there's 2 places where we want it to say rosa now instead of aws from what i can tell. one you fixed in the common file which will result in the metadata being correct. the second is by fixing the cloud_type variable that we were passing in previously in the env.sh file. if we just get rid of that var and pass the now correct PLATFORM var in its place i think we could simplify the entire thing and drop the get_cluster_type function
"

morenod commented on workloads/router-perf-v2/workload.py:
```diff
@@ -66,23 +65,31 @@ def run_mb(mb_config, runtime, output):
         return unknown_value
-    aws_data = platform_status_json["aws"]
-    if "resourceTags" not in aws_data:
+    if "type" in platform_status_json:
```
Updated

dry923 commented:
/rerun all
